File: 492b-vanya-and-lanterns/lanterns.py
n, length = map(int, input().split())

positions = list(map(int, input().split()))
positions.sort()

# print(*positions)

radii = [(positions[i] - positions[i - 1]) / 2 for i in range(1, n)]
radii += [positions[0], length - positions[-1]]

print(max(radii))

File: 492b-vanya-and-lanterns/lanterns.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main()
{
    int n, length;

    cin >> n >> length;

    vector<int> positions;

    for (int i = 0; i < n; i++)
    {
        int p;
        cin >> p;

        positions.push_back(p);
    }

    sort(positions.begin(), positions.end());

    double radius = max(positions[0], length - positions[n - 1]);

    for (int i = 1; i < n; i++)
    {
        radius = max(radius, (positions[i] - positions[i - 1]) / 2.0);
    }

    cout << std::fixed << radius << endl;
}
